sql/catalyst/src/main/scala/org/apache/spark/sql/sources/v2/filters.scala

```diff
@@ -0,0 +1,222 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.sources.v2
+
+import org.apache.spark.annotation.Experimental
+import org.apache.spark.sql.connector.expressions.NamedReference
+
+/**
+ * A filter that evaluates to `true` iff the field evaluates to a value
+ * equal to `value`.
+ *
+ * @since 3.0.0
+ */
+@Experimental
+case class EqualTo(ref: NamedReference, value: Any) extends FilterV2 {
+  override def references: Array[NamedReference] = Array(ref) ++ findReferences(value)
```

Comment:
Why is value `Any`? Shouldn't it be an expression (like the v2 `Literal`)?

Also, in Iceberg expressions we've updated `ref` to be a `Term` instead of a `Reference`. Both `Reference` and `Transform` are terms, which allows us to express that the value of a transformed reference is equal to something. That gives us the ability to express `date(ts) = '2020-01-17'`, for example.